### ISSUE TITLE
Create a routeable road network using the NRN.

### ISSUE THREAD

**Issue by ktjaco** [goatsweater/nrn-rrn#25]
_Is your feature request related to a problem? Please describe._

There is a possible need for point to point routing using the National Road Network.

_Describe the solution you'd like._

A routing API that takes coordinates to find the most optimal route depending on the question being asked.

_Describe alternatives you've considered._

One solution could be to use pgRouting and GeoServer. A simple application can be found at https://github.com/ktjaco/nrn-routing.

**Comment by goatsweater** [comment]
I agree that such a service would be valuable. It isn't clear to me where the responsibility for such a service would lie in terms of hosting the service. What I mean by this is maybe the NRN feeds into such a service, pushing updates into the backend and such, but then hosting that service is the responsibility of some other group?

I think there's benefit to having a portable system like you built. A user could run this to implement their own desktop based solution, but also someone with more resources could host it as a service that could live alongside the existing WMS and some future WFS. Ideally all the data would live in a single database to serve the WMS, WFS, and routing functions to make sure that it is always referencing the same data and giving consistent results.
